src/System.Linq/src/System/Linq/ThrowHelper.cs

```diff
@@ -0,0 +1,56 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Linq
+{
+    internal static class ThrowHelper
+    {
+        internal static void ThrowArgumentNullException(ExceptionArgument argument) => throw CreateArgumentNullException(argument);
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
The two layers (Throw+Create) are not needed with current RyuJIT.

Reply:
Ok. Most of these I did this way because the Create methods are also used in one of two places elsewhere, but I can remove it for this one case where it's not.

Reply:
Why does coreclr's corelib still use the two layers?
https://github.com/dotnet/coreclr/blob/e3f58a3d91f49c6d50aee2068aa885fe74428713/src/System.Private.CoreLib/shared/System/ThrowHelper.cs#L165-L173

Reply:
> Why does coreclr's corelib still use the two layers?

It uses them in some places, but not others. It is left over from times when the two layers are necessary.

The most efficient pattern for these boilerplate ThrowHelpers should be tool generated ... hope we will get these one day.

Reply:
> The most efficient pattern for these boilerplate ThrowHelpers should be tool generated ... hope we will get these one day.

I agree.  I actually debated not doing this specifically because of that, but since System.Linq was already using a pattern, a tool would almost certainly not refactor it, and as the pattern's been superseded, seemed to make sense to update it.